test/unit/audio-bandwidth-test.js

```diff
@@ -0,0 +1,351 @@
+  delete global.RTCPeerConnection;
+  delete global.navigator;
+  delete global.document.documentElement.style.WebkitAppearance;
+});
+
+test('start() return error if iceConfig has no iceServers', async t => {
+  try {
+    await audioBandwidthTest.start();
+  } catch (err) {
+    t.is(err.message, 'No ice servers were provided');
+  }
+});
+
+test('start() should call doGetUserMedia if there is iceServers and return error with results', async t => {
+  const options = {
+    options: {
+      mediaOptions: {
+        audio: {
+          deviceId: 'someAudioId'
+        }
+      },
+      iceConfig: {
+        iceServers: [{ server: '1' }]
+      }
+    },
+    doGetUserMedia: () => Promise.resolve(document.createElement('audio')),
+    runTest: () => {},
+    completed: () => {},
+    getResults: () => ['Some Audio'],
+    mediaOptions: {
+      audio: {
+        deviceId: 'someAudioId'
+      }
+    },
+    iceConfig: {
+      iceServers: [{ server: '1' }]
+    },
+    reject: () => 'return results and err'
+  };
+  const audioBandwidthTest = new AudioBandwidthTest(options);
+  // Mock out RTCPeerConnection for node runtime.
+  global.RTCPeerConnection = () => {
+    return {
+      addEventListener: () => {}
+    };
+  };
+  const actual = await audioBandwidthTest.start.call(options);
+  t.is(actual, 'return results and err');
+});
+
+test('getResults() should return object with log, constraints, and stats', t => {
+  const actual = audioBandwidthTest.getResults();
+  const expected = {
+    log: [],
+    stats: {},
+    constraints: { video: false, audio: { deviceId: 'someAudioId' } }
+  };
+  t.deepEqual(actual, expected);
+});
+
+test('addLog() should push message to the log', t => {
+  t.plan(2);
+  t.notThrows(() => audioBandwidthTest.addLog('info', { val: 'Test add log' }));
+  t.notThrows(() => audioBandwidthTest.addLog('error', 'my error'));
+});
+
+test('doGetUserMedia() should return stream', async t => {
+  const options = {
+    options: {
+      mediaOptions: {
+        audio: {
+          deviceId: 'someAudioId'
+        }
+      },
+      iceConfig: {
+        iceServers: [{ server: '1' }]
+      }
+    },
+    doGetUserMedia: () => Promise.resolve(document.createElement('audio')),
+    runTest: () => {},
+    completed: () => {},
+    getResults: () => ['Some Audio'],
+    mediaOptions: {
+      audio: {
+        deviceId: 'someAudioId'
+      }
+    },
+    iceConfig: {
+      iceServers: [{ server: '1' }]
+    },
+    reject: () => 'return results and err'
+  };
+  const audioBandwidthTest = new AudioBandwidthTest(options);
+  global.navigator = {
+    mediaDevices: {
+      getUserMedia: (constraints) => Promise.resolve(constraints)
+    }
+  };
+  let audioElement = document.createElement('audio');
+  audioElement.src = `data:audio/x-wav;base64,${new Buffer('wave')}>`;
+  audioElement.getAudioTracks = () => ['track1', 'track2'];
+  const actual = await audioBandwidthTest.doGetUserMedia(audioElement);
+  t.truthy(actual);
```

Comment:
Instead of just `truthy` could we ensure it's the same object resolved from the getUserMedia call, or the value we expect?